routes/scheduleRoutes.js

```diff
@@ -0,0 +1,94 @@
+const router = require("express").Router();
+
+// Import data models
+const db = require("../data/models");
+
+// Load middleware
+
+/**
+ * Method to retrieve all event schedules from the database
+ * @returns sends all schedules in the database as a response
+ */
+
+router.get("/", async (req, res) => {
+	try {
+		const data = await db.findAll("Schedules");
+		res.send(data);
+	} catch (err) {
+		res.status(500).send(err.message);
+	}
+});
+
+/**
+ * Method to retrieve a specific schedule from the database
+ * @param {number} scheduleId
+ * @returns sends the specific schedule as a response
+ */
+router.get("/:scheduleId", async (req, res) => {
+	const { scheduleId } = req.params;
+	try {
+		const data = await db.findById("Schedules", scheduleId);
+		if (data) {
+			res.send(data);
+		} else {
+			res.status(404).json({ message: `Record ${scheduleId} not found` });
+		}
+	} catch (err) {
+		res.status(500).send(err.message);
+	}
+});
+
+/**
+ * Method to add one schedule to the database
+ * @param {object} req.body
+ * @returns newly created record
+ */
+router.post("/", async (req, res) => {
+	try {
+		const data = await db.addRecord("Schedules", req.body);
```

Comment:
Just a thought for a different approach here. Do we want to validate the user req body prior to adding or updating a db record? I think we can use a single middleware to check for required fields across all post / put requests on our api.  Something like validateBody(table, body) -- and we could use an object with keys of table names and values are arrays of expected properties.  Just an idea.

Reply:
@louismagdaleno 

I totally agree. I have it on my list to iterate on these endpoints with data validation middleware. In previous projects I've created high order functions to easily assign and modify data requirements with each route.

Here's an example that I did for our last Sprint Challenge:
https://github.com/wmemorgan/Sprint-Challenge--Testing/blob/wilfred-morgan/routes/gamesRoutes.js